librz/asm/asm.c

```diff
@@ -304,13 +305,54 @@ RZ_API bool rz_asm_use_assembler(RzAsm *a, const char *name) {
 	return false;
 }
 
+/**
+ * \brief Copies all config nodes in \p pcfg to the config in \p rz_asm.
+ *
+ * \param rz_asm Pointer to RzAsm struct.
+ * \param pcfg Pointer to the plugins RzConfig struct.
+ */
+static void set_plugin_configs(RZ_BORROW RzAsm *rz_asm, RZ_BORROW RzConfig *pcfg) {
+	rz_return_if_fail(pcfg && rz_asm);
+
+	RzConfig *conf = ((RzCore *)(rz_asm->core))->config;
+	RzConfigNode *n;
+	RzListIter *it;
+	rz_list_foreach_iter(pcfg->nodes, it) {
+		n = it->data;
+		if (!rz_config_add_node(conf, rz_config_node_clone(n))) {
+			RZ_LOG_WARN("Failed add %s to the global config.\n", n->name)
+		}
+	}
+}
+
+/**
+ * \brief Deletes all copies of \p pcfg nodes in the RzConfig from \p rz_asm.
+ *
+ * \param rz_asm Pointer to RzAsm struct.
+ * \param pcfg Pointer to the plugins RzConfig struct.
+ */
+static void unset_plugins_config(RZ_BORROW RzAsm *rz_asm, RZ_BORROW RzConfig *pcfg) {
+	rz_return_if_fail(pcfg && rz_asm);
```

Comment:
shouldn't core also be checked?

Reply:
Resolved in https://github.com/rizinorg/rizin/pull/2488